discovery-infra/assisted_service_api.py

```diff
@@ -14,9 +18,51 @@ def __init__(self, inventory_url):
         self.inventory_url = inventory_url
         configs = Configuration()
         configs.host = self.inventory_url + "/api/assisted-install/v1"
+        if 'OFFLINE_TOKEN' in os.environ:
+            self.set_config_auth(configs)
+
         self.api = ApiClient(configuration=configs)
         self.client = api.InstallerApi(api_client=self.api)
 
+    def set_config_auth(self, c):
+        offline_token = os.environ.get('OFFLINE_TOKEN')
+        if offline_token == None: return
+
+        def refresh_api_key(config):
+            # Get the properly padded key segment
+            auth = config.api_key.get('Authorization', None)
+            if auth != None:
+                segment = auth.split('.')[1]
+                padding = len(segment) % 4
+                segment = segment + padding * '='
+
+                expires_on = json.loads(base64.b64decode(segment))['exp']
+
+                # if this key doesn't expire or if it has more than 30 seconds left, don't refresh
+                remaining = expires_on - time.time()
+                if expires_on == 0 or remaining > 30:
+                    return
+
+            # fetch new key if expired or not set yet
+            token_url = "https://sso.redhat.com/auth/realms/redhat-external/protocol/openid-connect/token"
+            params = {
+                "client_id":     "cloud-services",
+                "client_secret": offline_token,
```

Comment:
I noticed that in [pyocm](https://github.com/TheDiemer/pyocm/blob/master/pyocm/pyocm.py#L36-L39) they don't use `client_secret`, why is the delta?

Reply:
Great question. I took this from what I saw in `ocm token -v10` I'll try without the secret. It did seem strange to specify it twice.

Reply:
Yup seems to work without the secret for me using `curl` :+1: good catch.